src/main/java/com/cloudbees/plugins/credentials/GlobalCredentialsConfiguration.java

```diff
@@ -164,7 +165,7 @@ private boolean configure(StaplerRequest req, JSONObject json) throws FormExcept
 
         // persist all the provider configs
         boolean result = true;
-        for (Descriptor<?> d : Functions.getSortedDescriptorsForGlobalConfig(FILTER)) {
+        for (Descriptor<?> d : Functions.getSortedDescriptorsForGlobalConfigByDescriptor(Category.class::isInstance)) {
```

Comment:
does not compile

Reply:
you need to also

```diff
diff --git Jenkinsfile Jenkinsfile
index 87a086d..a229fa5 100644
--- Jenkinsfile
+++ Jenkinsfile
@@ -1 +1 @@
-buildPlugin(configurations: buildPlugin.recommendedConfigurations())
+buildPlugin()
```

Reply:
It compiles. Is it not supposed to?

Reply:
The current `Jenkinsfile` tries to build the plugin against an _older_ version of Jenkins, which will not work.

Reply:
If you have write permission to the plugin, you can make the above change. Otherwise you need to enlist the help of someone who does.

Reply:
Fixed in https://github.com/jenkinsci/credentials-plugin/pull/157